app.py

```diff
@@ -20,87 +16,66 @@ def on_run_app():
         beers = {}
 
 
-def on_exit_app():
+@app.on_event("shutdown")
+def shutdown_event():
     with open("data.json", "w+") as outfile:
         json.dump(beers, outfile)
 
 
-atexit.register(on_exit_app)
+@app.get("/beers/{isbn}")
+async def get_beer(isbn: str):
+    if isbn in beers:
+        return beers[isbn]
+    raise HTTPException(status_code=404, detail="Item not found")
 
 
-@app.route('/beers')
-def list_beer():
+@app.get("/beers")
+async def list_beer(req: Request):
     beer_list = list(beers.values())
     for filter_name in ['name', 'brand', 'isbn', 'price']:
-        filter_value = request.args.get(f'filter[{filter_name}]')
-
-        # if filter not given, nothing to filter, continue
-        if not filter_value:
-            continue
-
-        # do the filtering
-        filtered_beer_list = []
-        filter_values = filter_value.split(',')
-        for _beer in beer_list:
-            if _beer[filter_name] in filter_values:
-                filtered_beer_list.append(_beer)
-
-        # filtered list is now the "main" list
-        beer_list = filtered_beer_list
-
-    return jsonify(beer_list)
-
-
-@app.route('/beers', methods=['POST'])
-def create_beer():
-    if validate_beer_object():
+        if f'filter[{filter_name}]' in req.query_params:
+            filter_value = req.query_params[f'filter[{filter_name}]']
```

Comment:
DRY - https://en.wikipedia.org/wiki/Don%27t_repeat_yourself (doing the `f'filter[{filter_name}]'` twice)
One of
```suggestion
        filter_value = req.query_params.get(f'filter[{filter_name}]')
        if filter_value is not None:
```
or
```suggestion
        filter_query_param_name = if f'filter[{filter_name}]'
        if filter_query_param_name in req.query_params:
            filter_value = req.query_params[filter_query_param_name]
```